fairlearn/metrics/_roc_auc.py

```diff
@@ -0,0 +1,328 @@
+
+    def plot_by_group(
+        self,
+        sensitive_index=None,
+        title = None,
+        ax=None,
+        include_overall=True,
+        include_baseline=True,
+        pos_label=1,
+        **kwargs
+                ):
+        """
+        Plots ROC curve by sensitive feature subgroup.
+
+        To enable sub-groupings of sensitive features to be plotted separately
+        and allow for more readable plots, the user may select specific features
+        to plot. The object the user passes must be a Series index as returned
+        by the `by_group` method. By default the legend labels will be the name
+        included in the index for each sensitive feature subgrouping in the form
+        of a tuple.
+
+        The display of the plot can be customized by passing an axes object to
+        the `ax` parameter.
+
+        Parameters
+        ----------
+
+        sensitive_index : Optional Pandas Series object that contains the data
+            associated with the sensitive feature indexed by subgroup. If not
+            provided, default is to include all groups in the series index.
+
+        title : Optional title to add to the resulting figure. Typically the
+            name of the estimator used for classification
+            (i.e. 'Logistic Regression').
+
+        ax : A Matplotlib axes object.
+
+        include_overall : Whether to include a plot of the ROC curve for the
+            data as a whole for comparison. Default is true.
+
+        include_baseline : Whether to include a baseline (i.e. selection at
+            random) for comparison. Default is true.
+
+        pos_label : The label of the positive class. When pos_label=None, if
+            y_true is in {-1, 1} or {0, 1}, pos_label is set to 1, otherwise an
+            error will be raised.
+
+        **kwargs : Optional additional key word arguments to pass to the
+            Scikit-learn method roc_curve.
+
+        Returns
+        -------
+        ax : The Matplotlib axes object with data values added to the axes. If
+            using the default, the axes will also include the default legend
+            label (i.e. "Overall (AUC = 0.92)")
+
+        """
+        # Establish sensitive features
+        if not sensitive_index:
+            sensitive_index = self.sensitive_series.index
+
+        # Generate plot figure, ax if not provided
+        if not ax:
+            figure, ax = plt.subplots()
+
+        # Set plot title to name of estimator
+        if title:
+            ax.set_title(title)
+
+        # Plot baseline
+        if include_baseline:
+            ax = self.plot_baseline(ax=ax)
+
+        # Plot overall
+        if include_overall:
+            ax = self.plot_overall(ax=ax)
+
+        # Plot ROC Curves by group
+        for name in sensitive_index:
+            grp_y_true = self.sensitive_series[name][0]
+            grp_y_score = self.sensitive_series[name][1]
+            display = self.plot_roc(
```

Comment:
Is it possible to use `self.plot_roc` as the metric itself, with judicious use of `functools.partial` to bind the other arguments (not sure if that works with `**kwargs`).

Reply:
@riedgar-ms  Yeah, I tried that at first. It may be possible, but when I tried it I ran into some issues with the MetricFrame class because it computes the metric on instantiation, and all I am actually using from that class is the group by function to split the data. It was far easier to pass a placeholder function. Open to suggestions. However, I think the work that needs to happen is some small tweaks to MetricFrame..So holding out for us to determine the approach to MetricFrame and then we can come back to this and refactor it if needed.